controllers/controllers/workloads/cforg_controller.go

```diff
@@ -0,0 +1,172 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package workloads
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	workloadsv1alpha1 "code.cloudfoundry.org/cf-k8s-controllers/controllers/apis/workloads/v1alpha1"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/hierarchical-namespaces/api/v1alpha2"
+)
+
+// CFOrgReconciler reconciles a CFOrg object
+type CFOrgReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+	Log    logr.Logger
+}
+
+const (
+	StatusConditionReady  = "Ready"
+	OrgNameLabel          = "cloudfoundry.org/org-name"
+	hierarchyMetadataName = "hierarchy"
+	APIVersion            = "workloads.cloudfoundry.org/v1alpha1"
+	Kind                  = "CFOrg"
+)
+
+//+kubebuilder:rbac:groups=workloads.cloudfoundry.org,resources=cforgs,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=workloads.cloudfoundry.org,resources=cforgs/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=workloads.cloudfoundry.org,resources=cforgs/finalizers,verbs=update
+
+//+kubebuilder:rbac:groups="",resources=namespaces,verbs=get;list;watch
+//+kubebuilder:rbac:groups=hnc.x-k8s.io,resources=subnamespaceanchors,verbs=list;create;delete;watch
+//+kubebuilder:rbac:groups=hnc.x-k8s.io,resources=hierarchyconfigurations,verbs=get;list;watch;update;patch
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the CFOrg object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.11.0/pkg/reconcile
+func (r *CFOrgReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	cfOrg := new(workloadsv1alpha1.CFOrg)
+	err := r.Client.Get(ctx, req.NamespacedName, cfOrg)
+	if err != nil {
+		r.Log.Error(err, fmt.Sprintf("Error when trying to fetch CFOrg %s/%s", req.Namespace, req.Name))
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	if cfOrg.ObjectMeta.DeletionTimestamp != nil && !cfOrg.ObjectMeta.DeletionTimestamp.IsZero() {
+		return ctrl.Result{}, nil
+	}
+
+	var anchor v1alpha2.SubnamespaceAnchor
+	err = r.Client.Get(ctx, req.NamespacedName, &anchor)
+	if err != nil {
+		if !k8serrors.IsNotFound(err) {
+			r.Log.Error(err, fmt.Sprintf("Error getting SubnamespaceAnchor for CFOrg %s/%s", req.Namespace, req.Name))
+			return ctrl.Result{}, err
+		}
+
+		anchor = v1alpha2.SubnamespaceAnchor{
```

Comment:
It looks like a good idea to extract the code from this line until the end of the if to a private function, e.g. `createSubnamespaceAnchor`. This would make the code easier to read.

Reply:
Good call. Done